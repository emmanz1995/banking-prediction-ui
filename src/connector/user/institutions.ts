import connector from '../connector';

// ${import.meta.env.VITE_API_URL}

export const onSearchInstitutions = async (text: string) =>
  await connector(`http://localhost:8082/api/v1/institutions/search`, {
    method: 'POST',
    data: { text },
  });

export const getAllInstitutions = async (page: number, limit: number) =>
  await connector(
    `http://localhost:8082/api/v1/institutions?page=${page}&limit=${limit}`,
    {
      method: 'GET',
    }
  );
