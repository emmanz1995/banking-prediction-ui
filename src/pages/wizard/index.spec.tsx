import {render, fireEvent, waitFor} from '@testing-library/react';
import '@testing-library/jest-dom';
import AccessAccountsWizard from './index';
import { onSearchInstitutions } from '../../connector/user/institutions';

jest.mock('../../connector/user/institutions', () => ({
  onSearchInstitutions: jest.fn(),
}));

describe('test', () => {
  const mockInstitutions = [
    {
      _id: 'REVOLUT_REVOGB21',
      __v: 0,
      bic: 'REVOGB21',
      countries: ['GB'],
      createdAt: '2024-10-16T18:21:25.508Z',
      logo: 'https://storage.googleapis.com/gc-prd-institution_icons-production/UK/PNG/revolut.png',
      name: 'Revolut',
      transaction_total_days: '730',
      updatedAt: '2024-10-16T22:30:57.972Z',
    },
  ];
  it('should render component', async () => {
    const { container } = render(<AccessAccountsWizard />)

    expect(container).toMatchSnapshot()
  });

  it('should render initial layout and button', () => {
    const { getByText, getByRole } = render(<AccessAccountsWizard />);

    // Check that the title and button are rendered
    expect(getByText('Access Your Account')).toBeInTheDocument();
    expect(getByRole('button', { name: /institutions/i })).toBeInTheDocument();
  });

  it('should show modal after clicking the institutes button', async () => {
   // mocks the institutions
   (onSearchInstitutions as jest.MockedFunction<typeof onSearchInstitutions>).mockResolvedValue(mockInstitutions)
    const { getByRole, getByTestId, getByPlaceholderText, getByText, queryByText } = render(<AccessAccountsWizard />);

    // gets the button institution button
    const btn = getByRole('button');
    fireEvent.click(btn)

    const form = getByTestId('institution-search')
    const input = getByPlaceholderText('Search an institution')

    // fires on change event
    fireEvent.change(input, { target: { value: 'Revolut' } })

    await waitFor(() => expect(getByText('Loading...')).toBeInTheDocument())
    await waitFor(() => expect(queryByText('Loading...')).not.toBeInTheDocument())

    // asserting that these values exists in the documents
    expect(btn).toBeInTheDocument()
    expect(form).toBeInTheDocument()
    expect(input).toBeInTheDocument()
    expect(input).toHaveValue('Revolut')

    const closeBtn = getByRole('button', { name: /Close/i })
    fireEvent.click(closeBtn);

    expect(form).not.toBeInTheDocument();
    expect(input).not.toBeInTheDocument();
  })
});