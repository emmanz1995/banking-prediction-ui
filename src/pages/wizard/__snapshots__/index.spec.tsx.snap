// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test should render component 1`] = `
<div>
  <main
    class="sc-aXZVg ljoJvr"
  >
    <div>
      <div
        class="navbar"
      >
        <div
          class="title"
        >
          <h3>
            Hello World!
          </h3>
        </div>
      </div>
    </div>
    <div
      class="layout-header"
    >
      <div
        class="sidebar-left"
      >
        <div
          class="sidebar"
        >
          <div
            class="title"
          />
          <ul
            class="list"
          >
            <li>
              <a
                href="/dashboard"
              >
                Dashboard
              </a>
            </li>
            <li>
              <a
                href="/users"
              >
                Users
              </a>
            </li>
            <li>
              <a
                href="/profile"
              >
                Profile
              </a>
            </li>
            <li>
              <a
                href="/announcements"
              >
                Announcements
              </a>
            </li>
            <li>
              <a
                href="/about"
              >
                About
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        style="width: 100%;"
      >
        <div
          class="sc-dAlyuH glGfOt"
        >
          <div
            class="sc-cwHptR bdpYHD"
          >
            <div
              class="sc-jlZhew bObgCL"
            >
              <h3>
                Access Your Account
              </h3>
              emmanz95
            </div>
             
            <div>
              <button
                class="sc-gEvEer fEedjk"
              >
                Institutions
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
